components/homme/src/theta-l/share/element_ops.F90

```diff
@@ -75,7 +75,12 @@ module element_ops
   public state0
 
   ! promote this to _real_kind after V2 code freeze
+  ! For BFB testing, use double precision
+#ifdef HOMMEXX_BFB_TESTING
```

Comment:
Based on discussion in #5286, you shouldn't need this ifdef because tref_lapse_rate does not affect anything except HOMME standalone tests; that is, this is not a non-BFB change w.r.t. E3SM v2 testing. Thus, just make the original line use d0.